### ISSUE TITLE
DockerSpawner template_namespace

### ISSUE THREAD

**Issue by DeoLeung** [jupyterhub/dockerspawner#290]
```bash
[I 2019-03-03 06:25:03.970 JupyterHub app:1673] Using Authenticator: oauthenticator.gitlab.GitLabOAuthenticator-0.8.1
[I 2019-03-03 06:25:03.971 JupyterHub app:1673] Using Spawner: dockerspawner.dockerspawner.DockerSpawner-0.11.0
[E 2019-03-03 06:25:30.635 JupyterHub pages:148] Failed to spawn single-user server with form
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/handlers/pages.py", line 146, in post
        await self.spawn_single_user(user, options=options)
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/handlers/base.py", line 705, in spawn_single_user
        timedelta(seconds=self.slow_spawn_timeout), finish_spawn_future
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/handlers/base.py", line 626, in finish_user_spawn
        await spawn_future
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/user.py", line 489, in spawn
        raise e
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/user.py", line 409, in spawn
        url = await gen.with_timeout(timedelta(seconds=spawner.start_timeout), f)
      File "/opt/conda/lib/python3.6/site-packages/dockerspawner/dockerspawner.py", line 984, in start
        obj = yield self.create_object()
      File "/opt/conda/lib/python3.6/site-packages/dockerspawner/dockerspawner.py", line 852, in create_object
        command=(yield self.get_command()),
      File "/opt/conda/lib/python3.6/site-packages/dockerspawner/dockerspawner.py", line 812, in get_command
        return cmd + self.get_args()
      File "/opt/conda/lib/python3.6/site-packages/dockerspawner/dockerspawner.py", line 726, in get_args
        args = super().get_args()
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/spawner.py", line 700, in get_args
        notebook_dir = self.format_string(self.notebook_dir)
      File "/opt/conda/lib/python3.6/site-packages/jupyterhub/spawner.py", line 681, in format_string
        return s.format(**self.template_namespace())
    TypeError: 'dict' object is not callable
```

the `template_namespace` is overwritten as a property

**Comment by flixr** [comment]
I just ran into the same error, seems this was introduced in #267

**Comment by vlizanae** [comment]
same

**Comment by mathematicalmichael** [comment]
also running into this... VERY frustrating. will this be fixed?

**Comment by playermanny2** [comment]
Same issue here! RUN pip install dockerspawner==0.10.0 inside my Dockerfile as a temporary solutino

**Comment by manniche** [comment]
in https://github.com/jupyterhub/jupyterhub/blob/0.9.6/jupyterhub/spawner.py#L699 and forward, the code checks if the configuration values for `notebook_dir` or `default_url` are set. If either is the case, it will call `s.format` as a method, when it at runtime is a `dict` type. Having either `c.Spawner.default_url` or c.DockerSpawner.notebook_dir` in your `jupyterhub_config.py` will trigger this error.

**Comment by mathematicalmichael** [comment]
@manniche but if `c.Spawner.default_url` or `c.DockerSpawner.notebook_dir` are commented out of the config, how does the hub know what default directory to launch in to? Is this just a bug waiting to be fixed?
Thanks for the explanation!

**Comment by minrk** [comment]
@cnydw thanks for the fixes! Will merge 0.11.1 with them shortly.
